src/Microsoft.AspNetCore.Sockets/Transports/LongPollingTransport.cs

```diff
@@ -37,18 +37,19 @@ public LongPollingTransport(IReadableChannel<Message> connection, ILoggerFactory
                 // TODO: We need the ability to yield the connection without completing the channel.
                 // This is to force ReadAsync to yield without data to end to poll but not the entire connection.
                 // This is for cases when the client reconnects see issue #27
-                using (var message = await _connection.ReadAsync(context.RequestAborted))
+                await _application.WaitToReadAsync(context.RequestAborted);
```

Comment:
Why would it 404? TryRead would return false no? It would return 200 and the next request would get the 204 because the channel is complete. We could optimize it by returning the 204 immediately if WaitToReadAsync returns false.